### PR TITLE
Support additional mkdocs server cli params

### LINKED ISSUE #21370: 🚀 Feature: [Techdocs-cli] allow additional mkdocs command-line parameters
### 🔖 Feature description

Would be nice to enhance the techdocs-cli with an additional command-line parameter to pass through mkdocs command-lines parameter

For example:
`techdocs-cli serve --mkdocs-config "--clean"` would pass through `--clean` to the mkdocsServer and start mkdocs like this:
```bash
mkdocs serve --clean
```

### 🎤 Context

Techdocs-cli uses `mkdocs serve` to build your docs locally. In some cases we have the need for additional mkdocs command-line parameters like `--clean` or `--strict`, e.g. to validate your mkdocs before pushing it to git or `--clean` to be able to [exclude docs](https://www.mkdocs.org/user-guide/configuration/#exclude_docs)

### ✌️ Possible Implementation

Add the additional config parameter here: https://github.com/backstage/backstage/blob/master/packages/techdocs-cli/src/lib/mkdocsServer.ts#L72 
and here: https://github.com/backstage/backstage/blob/master/packages/techdocs-cli/src/lib/mkdocsServer.ts#L55

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

### PR DESCRIPTION
**Pull request by hainenber**
#### :heavy_check_mark: Checklist
Partially fixes #21370 and #20972

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by awanlin — CHANGES_REQUESTED** [review]
Hi @hainenber, overall this looks good but before I do a deep review would you be open to a name change? Using `--mkdocs-configs` is confusing, could this be called `--mkdocs-parameters` instead? That seems more clear. If that works for you then can you update the code to use the new name as well, please?

**Review by agentbellnorm — CHANGES_REQUESTED** [review]
Thanks for the PR, It makes sense to add this capability!

I understand the thinking behind being able to pass any parameter, but I'd like to be a bit conservative in what features we enable. Being able to pass any mkdocs flag suggests that techdocs supports all mkdocs flags, which is not necessarily true. Can you change your PR to only map the --dirty flag?

Here's a similar PR: https://github.com/backstage/backstage/pull/20356/files

**Comment by hainenber** [comment]
> Thanks for the PR, It makes sense to add this capability!
> 
> I understand the thinking behind being able to pass any parameter, but I'd like to be a bit conservative in what features we enable. Being able to pass any mkdocs flag suggests that techdocs supports all mkdocs flags, which is not necessarily true. Can you change your PR to only map the --dirty flag?
> 
> Here's a similar PR: https://github.com/backstage/backstage/pull/20356/files

Thanks for giving out the lead, @agentbellnorm. Appreciate it!

I understand the conservativeness for introducing new capacities so we can restrict the mkdocs flags. However, I'd like to support the folks's cases in linked issues, #21370 and #20972. Can we support these flags? `--dirtyreload`, `--dirty`, `--strict` and `--clean`?

**Comment by agentbellnorm** [comment]
> Can we support these flags? `--dirtyreload`, `--dirty`, `--strict` and `--clean`?

Yeah I see no problem being able to specify those flags! If you've tested them and they work fine, I'd be happy to review changes letting the user pass those.

**Comment by wtchnm** [comment]
Thanks for your work @hainenber! Just a note, the `--dirty` option is not available in currently used mkdocs version:

image

`techdocs cli serve` uses [`spotify/techdocs:v1.2.1`](https://github.com/backstage/backstage/blob/b70b16c5b669f217cbdb5c39d19b519ded0903c2/plugins/techdocs-node/src/stages/generate/techdocs.ts#L56), which uses `mkdocs@1.4.3`. The new name was only introduced in [`mkdocs@1.5.0`](https://www.mkdocs.org/about/release-notes/#version-150-2023-07-26).

File: .changeset/tricky-donkeys-do.md
Comment (awanlin) [inline .changeset/tricky-donkeys-do.md:5]: Can we either include the new parameters here or mention to run `techdocs-cli --help` for more details? This just helps Adopters learn a bit more about this change when they review the release (changesets get rollup into release notes)

**Review by awanlin — CHANGES_REQUESTED** [review]
This looks great, code is good, I left a comment regarding the changeset.

The only other outstanding item before I feel ready to approve is to resolve the comment @wtchnm mentioned: https://github.com/backstage/backstage/pull/21403#issuecomment-1820799981

File: .changeset/tricky-donkeys-do.md
Comment (hainenber) [reply .changeset/tricky-donkeys-do.md]: Yes, I've included details of new passable `mkdocs` params in latest commit. PTAL and thanks for the review!

File: packages/techdocs-cli/src/lib/mkdocsServer.ts
Comment (awanlin) [inline packages/techdocs-cli/src/lib/mkdocsServer.ts:62]:
```suggestion
        ...(options.mkdocsParameterDirtyReload ? ['--dirtyreload'] : []),
```

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for updating the changeset and the parameters, spotted one issue though

**Review by awanlin — APPROVED** [review]
🎉  Awesome!

**Comment by awanlin** [comment]
@agentbellnorm can you give this another look, please? Think's it's good to go unless you spot anything 👀

**Review by agentbellnorm — APPROVED** [review]
Good stuff! 🎸

**Pull request merged**
